src/praisonai-agents/praisonaiagents/agent/durable.py

```
@@ -0,0 +1,296 @@
+                function_name, arguments, tool_call_id
+            )
+            try:
+                result = execute_tool_fn(
+                    function_name,
+                    arguments,
+                    tool_call_id=tool_call_id,
+                    **kwargs,
+                )
+                if inspect.isawaitable(result):
+                    result = await result
+            except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)